### PR TITLE
[SMTChecker] Fix compound assignment to push

### LINKED ISSUE #11143: [SMTChecker] Incorrect analysis of subtract assignment to push
For the following example:
```
pragma experimental SMTChecker;
contract C {
	int[] u;
	function t() public {
		require(u.length == 0);
		u.push() -= 1;
		assert(u[0] < 0); // should hold
		assert(u[0] >= 0); // should fail
	}
}
```

SMTChecker incorrectly reports the first assertion as invalid (false positive) and the second assertion as valid (false negative).

### PR DESCRIPTION
**Pull request by blishko**
Fixes #11143.

### CODE REVIEW

File: libsolidity/formal/SMTEncoder.cpp [hunk 1]
```diff
@@ -2022,6 +1998,10 @@ void SMTEncoder::assignment(
 		dynamic_cast<MemberAccess const*>(left)
 	)
 		indexOrMemberAssignment(*left, _right);
+	else if (isEmptyPush(*left))
+	{
```
Comment (leonardoalt) [inline libsolidity/formal/SMTEncoder.cpp]: Can remove brackets

File: libsolidity/formal/SMTEncoder.cpp [hunk 2]
```diff
@@ -375,30 +375,9 @@ bool SMTEncoder::visit(Assignment const& _assignment)
...
-
-		arrayPushPopAssign(memberExpr, symbArray.currentValue());
-		defineExpr(_assignment, expr(left));
-		return false;
-	}
-
-	return true;
+	right.accept(*this);
+	left.accept(*this);
+	return false;
```
Comment (leonardoalt) [inline libsolidity/formal/SMTEncoder.cpp:379]: If all this function is doing now is visit left and right and returning false, I think we can remove it entirely, and only rely on `endVisit(Assignment)`

Comment (blishko) [reply libsolidity/formal/SMTEncoder.cpp:379]: I wanted to do that, but that does not work. The point is that we need to first visit RHS and then LHS, but the default is first LHS and then RHS.
I also checked `ExpressionCompiler` and that is also visiting assignment first RHS and then LHS.

Comment (leonardoalt) [reply libsolidity/formal/SMTEncoder.cpp:379]: Ah ok yes, missed that.

File: libsolidity/formal/SMTEncoder.cpp (hunk 1 above)
Comment (blishko) [reply libsolidity/formal/SMTEncoder.cpp]: Fixed.

File: libsolidity/formal/SMTEncoder.cpp (hunk 2 above)
Comment (blishko) [reply libsolidity/formal/SMTEncoder.cpp:379]: Added a comment explaining that.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**